src/driver.rs

```diff
@@ -366,3 +352,140 @@ impl MajorObject for Driver {
 }
 
 impl Metadata for Driver {}
+
+/// A wrapper around `GDALDriverManager`.
+/// This struct helps listing and registering [`Driver`]s.
+pub struct DriverManager;
+
+impl DriverManager {
+    /// Returns the number of registered drivers.
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// println!("{} drivers are registered", DriverManager::count());
+    /// ```
+    /// ```text
+    /// 203 drivers are registered
+    /// ```
+    pub fn count() -> usize {
+        _register_drivers();
+        let count = unsafe { gdal_sys::GDALGetDriverCount() };
+        count
+            .try_into()
+            .expect("The returned count should be zero or positive")
+    }
+
+    /// Returns the driver with the given index, which must be less than the value returned by
+    /// `DriverManager::count()`.
+    ///
+    /// See also: [`count`](Self::count)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// assert!(DriverManager::count() > 0);
+    /// let d = DriverManager::get_driver(0)?;
+    /// println!("'{}' is '{}'", d.short_name(), d.long_name());
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// 'VRT' is 'Virtual Raster'
+    /// ```
+    pub fn get_driver(index: usize) -> Result<Driver> {
+        _register_drivers();
+        let c_driver = unsafe { gdal_sys::GDALGetDriver(index.try_into().unwrap()) };
+        if c_driver.is_null() {
+            return Err(_last_null_pointer_err("GDALGetDriver"));
+        }
+        Ok(Driver { c_driver })
+    }
+
+    /// Returns the driver with the given short name or [`Err`] if not found.
+    ///
+    /// See also: [`count`](Self::count), [`get`](Self::get)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let cog_driver = DriverManager::get_driver_by_name("COG")?;
+    /// println!("{}", cog_driver.long_name());
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// Cloud optimized GeoTIFF generator
+    /// ```
+    pub fn get_driver_by_name(name: &str) -> Result<Driver> {
+        _register_drivers();
+        let c_name = CString::new(name)?;
+        let c_driver = unsafe { gdal_sys::GDALGetDriverByName(c_name.as_ptr()) };
+        if c_driver.is_null() {
+            return Err(_last_null_pointer_err("GDALGetDriverByName"));
+        };
+        Ok(Driver { c_driver })
+    }
+
+    /// Register a driver for use.
+    ///
+    /// Wraps [`GDALRegisterDriver()`](https://gdal.org/api/raster_c_api.html#_CPPv418GDALRegisterDriver11GDALDriverH)
+    pub fn register_driver(driver: &Driver) -> usize {
+        let index = unsafe { gdal_sys::GDALRegisterDriver(driver.c_driver) };
```

Comment:
It's not impossible, but it's pretty hard to end up needing to call this (because we do `_register_drivers` in `Get*`).